src/connectedmachine/azext_connectedmachine/tests/latest/test_connectedmachine_scenario.py

```diff
@@ -1,32 +1,221 @@
-# --------------------------------------------------------------------------------------------
+# --------------------------------------------------------------------------
 # Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License. See License.txt in the project root for license information.
-# --------------------------------------------------------------------------------------------
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is
+# regenerated.
+# --------------------------------------------------------------------------
 
 import os
-import unittest
-
+from azure.cli.testsdk import ScenarioTest
+from .. import try_manual, raise_if, calc_coverage
+from azure.cli.testsdk import ResourceGroupPreparer
 from azure_devtools.scenario_tests import AllowLargeResponse
-from azure.cli.testsdk import (ScenarioTest, ResourceGroupPreparer)
-
 
 TEST_DIR = os.path.abspath(os.path.join(os.path.abspath(__file__), '..'))
+CUSTOM_SCRIPT_EXTENSION_NAME = 'customScript'
+DEPENDENCY_AGENT_EXTENSION_NAME = 'dependencyAgent'
+
+# Constants for Machine tests
+MACHINES_RESOURCE_GROUP_NAME = 'AzcmagentTest'
```

Comment:
Normally we use a `ResourceGroupPreparer` in tests, it will clean up the resources even if the tests failed.

You can create and delete resource groups in the setup and cleanup too.

Reply:
Because Azure Arc for Servers onboarding process is different than normal Azure resources (there is no create because "create" involves downloading and running an special agent), all resources need to be created ahead of time so there is no need to use the Resource Group Preparer.